lua/chatcommands/coinflip.lua

```diff
@@ -0,0 +1,102 @@
+CoinFlips = CoinFlips or {}
+RegisterChatCommand({'coin','coinflip'},function(ply,arg)
+	if arg:lower() == "accept" then checkCoinFlipRequest(ply) return end
+	local t = string.Explode(" ",arg)
+	local p = tonumber(table.remove(t))
+	if p == nil then
+		ply:ChatPrint("[orange]!coinflip player points")
+	else
+		local to,c = PlyCount(string.Implode(" ",t))
+		if c == 1 then
+			if ply == to then
+				ply:ChatPrint("[red]You can't coinflip yourself!")
+			elseif p >= 1000 then -- minimum 1000 point coinflip
+				initCointFlip(ply,to,p)
+			else
+				ply:ChatPrint("[red]A coin flip must be a minimum of 1000 points. No pussy bets!")
+			end
+		else
+			ply:ChatPrint("[red]Player "..string.Implode(" ",t).." not found")
+		end
+	end
+end,{global=true,throttle=true})
+
+function initCointFlip(ply,target,amount)
+	if ply:PS_HasPoints(amount) and target:PS_HasPoints(amount) and CoinFlips[ply:SteamID()] == nil then
+		-- Both players have enough points AND an existing coinflip doesn't exist.
+		ply:ChatPrint("[orange]"..to:Nick().." is receiving your coinflip request.")
+		CoinFlips[ply:SteamID()] = {target:SteamID(), amount}
+		timer.Simple(120, CoinFlipRequestTimeout, ply, target) -- 2 minutes for the timeout.
```

Comment:
This timer never gets removed, so the request timeout thing will screw with later requests. And it will always print that stuff too.

Reply:
Totally forgot about that. Thanks for catching that. So the resolution is to make a non-simple timer with a unique identifier, and save the identifier in the CoinFlip?

Reply:
Something like `timer.Create( "CoinFlip" .. ply:SteamID(), 120, 1, CoinFlipRequestTimeout, ply, target)` and anywhere I do table.remove I will replace with a function that removes the table but also removes the timer?

Reply:
A more solid solution would be to use one global timer that runs every second `timer.Create("Coinflips",1,0, ...)` which checks each one in the table. Then have CurTime() written in each table member when you create it, and if the Coinflips timer sees something more than 60 seconds old, delete it

Reply:
Was thinking about doing this but didn't know if it was the right move or not. You say 60 seconds, is that the timeout you want?

Reply:
seems fine